frag-nav/src/main/java/com/ncapdevi/fragnav/FragNavTransactionOptions.java

```diff
@@ -0,0 +1,78 @@
+package com.ncapdevi.fragnav;
+
+import android.support.annotation.AnimRes;
+import android.support.annotation.StyleRes;
+import android.support.v4.util.Pair;
+import android.view.View;
+
+import java.util.List;
+
+/**
+ *
+ */
+
+
+public class FragNavTransactionOptions {
+    List<Pair<View, String>> sharedElements;
+    @FragNavController.Transit
+    Integer transition;
+    @AnimRes
+    Integer enterAnimationId;
+    @AnimRes
+    Integer exitAnimationId;
+    @StyleRes
+    Integer transitionStyle;
+
+    private FragNavTransactionOptions(Builder builder) {
+        sharedElements = builder.sharedElements;
+        transition = builder.transition;
+        enterAnimationId = builder.enterAnimationId;
+        exitAnimationId = builder.exitAnimationId;
+        transitionStyle = builder.transitionStyle;
+    }
+
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    public static final class Builder {
+        private List<Pair<View, String>> sharedElements;
+        private Integer transition;
+        private Integer enterAnimationId;
+        private Integer exitAnimationId;
+        private Integer transitionStyle;
+
+        private Builder() {
+        }
+
+        public Builder sharedElements(List<Pair<View, String>> val) {
```

Comment:
maybe add a builder method to add individual shared elements **in addition** to a list? Might make things easier:

```
new Builder()
  .sharedElement(view, string)
  .sharedElement(view, string)
```
vs
```
List<Pair<View, String>> sharedElements = new ArrayList<>();
sharedElements.add(Pair.create(view, string));
sharedElements.add(Pair.create(view, string));

new Builder()
   .sharedElements(sharedElements)
```

Can't hurt to have both.